src/Interpreters/ClusterProxy/executeQuery.cpp

```diff
@@ -206,15 +206,28 @@ ContextMutablePtr updateSettingsAndClientInfoForCluster(const Cluster & cluster,
     /// disable parallel replicas if cluster contains only shards with 1 replica
     if (context->canUseTaskBasedParallelReplicas())
     {
-        bool disable_parallel_replicas = true;
-        for (const auto & shard : cluster.getShardsInfo())
+        bool disable_parallel_replicas = false;
+        if (is_remote_function)
         {
-            if (shard.getAllNodeCount() > 1)
+            if (cluster.getName().empty()) // disable parallel replicas with remote() table functions w/o configured cluster
+                disable_parallel_replicas = true;
+            else
+                new_settings[Setting::cluster_for_parallel_replicas] = cluster.getName();
```

Comment:
What if the user specified `cluster_for_parallel_replicas` himself?

Reply:
`cluster_for_parallel_replicas` setting specified by user has no effect on `remote()` after this change

Reply:
It seems a bit like unexpected behaviour for user